File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/http/HttpServerResponse.php
<?php 
namespace io\vertx\php\core\http;
use io\vertx\php\core\MultiMap;
use io\vertx\php\core\buffer\Buffer;
class HttpServerResponse
{
    
    private function __construct()
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function bodyEndHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return num
     * methodSize : 1
     */
    public function bytesWritten(/*0*/)
    {

    }
    /**
     * ()
     * @return Void
     * methodSize : 1
     */
    public function close(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function closeHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function closed(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function drainHandler(/*1*/$arg0)
    {

    }
    /**
     * (string chunk)
     * @return Void
     * (Buffer chunk)
     * @return Void
     * (string chunk, string enc)
     * @return Void
     * ()
     * @return Void
     * methodSize : 4
     */
    public function end(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function endHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function ended(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return int
     * methodSize : 1
     */
    public function getStatusCode(/*0*/)
    {

    }
    /**
     * ()
     * @return string
     * methodSize : 1
     */
    public function getStatusMessage(/*0*/)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function headWritten(/*0*/)
    {

    }
    /**
     * ()
     * @return MultiMap
     * methodSize : 1
     */
    public function headers(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function headersEndHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isChunked(/*0*/)
    {

    }
    /**
     * (enum method, string host, string path, callable handler)
     * @return HttpServerResponse
     * (enum method, string path, MultiMap headers, callable handler)
     * @return HttpServerResponse
     * (enum method, string path, callable handler)
     * @return HttpServerResponse
     * (enum method, string host, string path, MultiMap headers, callable handler)
     * @return HttpServerResponse
     * methodSize : 4
     */
    public function push(/*5*/$arg0, $arg1, $arg2, $arg3, $arg4)
    {

    }
    /**
     * (string name, string value)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function putHeader(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (string name, string value)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function putTrailer(/*2*/$arg0, $arg1)
    {

    }
    /**
     * ()
     * @return Void
     * (num code)
     * @return Void
     * methodSize : 2
     */
    public function reset(/*1*/$arg0)
    {

    }
    /**
     * (string filename)
     * @return HttpServerResponse
     * (string filename, num offset)
     * @return HttpServerResponse
     * (string filename, callable resultHandler)
     * @return HttpServerResponse
     * (string filename, num offset, num length)
     * @return HttpServerResponse
     * (string filename, num offset, callable resultHandler)
     * @return HttpServerResponse
     * (string filename, num offset, num length, callable resultHandler)
     * @return HttpServerResponse
     * methodSize : 6
     */
    public function sendFile(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (bool chunked)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function setChunked(/*1*/$arg0)
    {

    }
    /**
     * (int statusCode)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function setStatusCode(/*1*/$arg0)
    {

    }
    /**
     * (string statusMessage)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function setStatusMessage(/*1*/$arg0)
    {

    }
    /**
     * (int maxSize)
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function setWriteQueueMaxSize(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return int
     * methodSize : 1
     */
    public function streamId(/*0*/)
    {

    }
    /**
     * ()
     * @return MultiMap
     * methodSize : 1
     */
    public function trailers(/*0*/)
    {

    }
    /**
     * (Buffer data)
     * @return HttpServerResponse
     * (string chunk)
     * @return HttpServerResponse
     * (string chunk, string enc)
     * @return HttpServerResponse
     * methodSize : 3
     */
    public function write(/*2*/$arg0, $arg1)
    {

    }
    /**
     * ()
     * @return HttpServerResponse
     * methodSize : 1
     */
    public function writeContinue(/*0*/)
    {

    }
    /**
     * (int type, int flags, Buffer payload)
     * @return HttpServerResponse
     * (HttpFrame frame)
     * @return HttpServerResponse
     * methodSize : 2
     */
    public function writeCustomFrame(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function writeQueueFull(/*0*/)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/http/HttpServerResponse.java
package io.vertx.php.core.http;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.buffer.Buffer;
import io.vertx.php.core.http.HttpFrame;
import io.vertx.php.core.streams.WriteStream;
import io.vertx.php.core.MultiMap;
@Name("HttpServerResponse")
@Namespace("io\\vertx\\php\\core\\http")
public interface HttpServerResponse<S extends io.vertx.core.http.HttpServerResponse> extends IWrapper<S>{
    
    @Signature
    default Memory writeQueueFull(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().writeQueueFull()
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }

    @Signature
    default Memory write(Environment __env__, Memory... args) {
        this.getWrappedObject().write(data)
    }

    @Signature
    default Memory setWriteQueueMaxSize(Environment __env__, Memory... args) {
        this.getWrappedObject().setWriteQueueMaxSize(maxSize)
    }

    @Signature
    default Memory drainHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().drainHandler(handler)
    }
    
    @Signature
    default Memory getStatusCode(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().getStatusCode()
    }

    @Signature
    default Memory setStatusCode(Environment __env__, Memory... args) {
        this.getWrappedObject().setStatusCode(statusCode)
    }
    
    @Signature
    default Memory getStatusMessage(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().getStatusMessage()
    }

    @Signature
    default Memory setStatusMessage(Environment __env__, Memory... args) {
        this.getWrappedObject().setStatusMessage(statusMessage)
    }

    @Signature
    default Memory setChunked(Environment __env__, Memory... args) {
        this.getWrappedObject().setChunked(chunked)
    }
    
    @Signature
    default Memory isChunked(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isChunked()
    }
    
    @Signature
    default Memory headers(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MultiMapImpl::new, this.getWrappedObject().headers()
    }

    @Signature
    default Memory putHeader(Environment __env__, Memory... args) {
        this.getWrappedObject().putHeader(name, value)
    }
    
    @Signature
    default Memory trailers(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MultiMapImpl::new, this.getWrappedObject().trailers()
    }

    @Signature
    default Memory putTrailer(Environment __env__, Memory... args) {
        this.getWrappedObject().putTrailer(name, value)
    }

    @Signature
    default Memory closeHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().closeHandler(handler)
    }

    @Signature
    default Memory endHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().endHandler(handler)
    }
    
    @Signature
    default Memory writeContinue(Environment __env__, Memory... args) {
        this.getWrappedObject().writeContinue()
    }
    
    @Signature
    default void end(Environment __env__, Memory... args) {
        this.getWrappedObject().end()
    }

    @Signature
    default void end(Environment __env__, Memory... args) {
        this.getWrappedObject().end()
    }

    @Signature
    default Memory sendFile(Environment __env__, Memory... args) {
        this.getWrappedObject().sendFile(filename)
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }
    
    @Signature
    default Memory ended(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().ended()
    }
    
    @Signature
    default Memory closed(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().closed()
    }
    
    @Signature
    default Memory headWritten(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().headWritten()
    }

    @Signature
    default Memory headersEndHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().headersEndHandler(handler)
    }

    @Signature
    default Memory bodyEndHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().bodyEndHandler(handler)
    }
    
    @Signature
    default Memory bytesWritten(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().bytesWritten()
    }
    
    @Signature
    default Memory streamId(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().streamId()
    }

    @Signature
    default Memory push(Environment __env__, Memory... args) {
        this.getWrappedObject().push(method, path, handler)
    }
    
    @Signature
    default void reset(Environment __env__, Memory... args) {
        this.getWrappedObject().reset()
    }

    @Signature
    default void reset(Environment __env__, Memory... args) {
        this.getWrappedObject().reset()
    }

    @Signature
    default Memory writeCustomFrame(Environment __env__, Memory... args) {
        this.getWrappedObject().writeCustomFrame(frame)
    }

}
